src/sdk/data-api.ts

```diff
@@ -29,6 +30,7 @@ const pkgPath = join(
 );
 const pkg = readFileSync(pkgPath, "utf8");
 const ClientVersion = JSON.parse(pkg).version;
+const knownBinaryFields = { ContentVersion: ["VersionData"] };
```

Comment:
I'm not sure what you mean. The data we get back from the API gives no real hint as to which fields might be binary. Seems like you have to check the docs to know. We've got a static list of known binary fields on the Java implementation too.

Reply:
if you describe the `ContentVersion` field for `VersionData` it does indicate that `type=base64` and `soapType=xsd:base64Binary`. 

```sh
sfdx force:schema:sobject:describe -s ContentVersion | jq '.fields[] | select(.name == "VersionData")'
```

And the description of the `base64` type [here](https://developer.salesforce.com/docs/atlas.en-us.236.0.object_reference.meta/object_reference/primitive_data_types.htm) states:

> Base 64-encoded binary data. Fields of this type are used for storing binary files in Attachment records, Document records, and Scontrol records. In these objects, the Body or Binary field contains the (base64 encoded) data, while the BodyLength field defines the length of the data in the Body or Binary field. In the Document object, you can specify a URL to the document instead of storing the document directly in the record.

which makes me wonder if the client should attempt to fetch and cache sObject metadata to use for driving some of this logic instead of having a hardcoded list.

Reply:
The java implementation is shipping with a static list too: 
https://github.com/forcedotcom/sf-fx-runtime-java/blob/51b288979acd40d560023a33a854010dd2bd7d20/sf-fx-runtime-java-sdk-impl-v1.1/src/main/java/com/salesforce/functions/jvm/runtime/sdk/BinaryFieldUtil.java#L22

Though, I do like the programatic approach for generating these. @Malax what do you think?

Reply:
@colincasey and I decided this would be a nice future improvement, and could also potentially add functionality where we have fully typed sObjects in later versions.